addons/pos_self_order_razorpay/static/src/app/self_order_service.js

```diff
@@ -0,0 +1,45 @@
+/** @odoo-module */
+import { patch } from "@web/core/utils/patch";
+import { SelfOrder } from "@pos_self_order/app/self_order_service";
+import { Razorpay, RazorpayError } from "@pos_self_order_razorpay/app/razorpay";
+
+patch(SelfOrder.prototype, {
+    async setup() {
+        await super.setup(...arguments);
+
+        const razorpayPaymentMethod = this.pos_payment_methods.find(
+            (p) => p.use_payment_terminal === "razorpay"
+        );
+
+        if (razorpayPaymentMethod) {
+            this.razorpay = new Razorpay(
+                this.env,
+                razorpayPaymentMethod,
+                this.access_token,
+                this.pos_config_id,
+                this.handleRazorpayError.bind(this),
+            );
+        }
+    },
+    handleReaderConnection(state) {
+        this.razorpayState = state.status;
+    },
+    handleRazorpayError(error, type) {
+        this.paymentError = true;
+        this.handleErrorNotification(error, type);
+    },
+    handleErrorNotification(error, type = "danger") {
+        let message = "";
+
+        if (error instanceof RazorpayError) {
+            message = `Razorpay POS: ${error}`;
+        } else {
+            super.handleErrorNotification(...arguments);
+            return;
+        }
+
+        this.notification.add(message, {
+            type: type,
+        });
```

Comment:
```suggestion
            this.notification.add(message, {
                type: type,
            });
        }
        super.handleErrorNotification(...arguments);
```
It should work this way, or you can keep the else block to call super but I don't think might be required